WordPress/Classes/Services/ReaderSearchSuggestionService.swift

```diff
@@ -4,22 +4,32 @@ import CocoaLumberjack
 /// Provides functionality for fetching, saving, and deleting search phrases
 /// used to search for content in the reader.
 ///
-@objc class ReaderSearchSuggestionService: LocalCoreDataService {
+@objc class ReaderSearchSuggestionService: NSObject {
+
+    private let coreDataStack: CoreDataStack
+
+    @objc init(coreDataStack: CoreDataStack) {
+        self.coreDataStack = coreDataStack
+        super.init()
+    }
 
     /// Creates or updates an existing record for the specified search phrase.
     ///
     /// - Parameters:
     ///     - phrase: The search phrase in question.
     ///
     @objc func createOrUpdateSuggestionForPhrase(_ phrase: String) {
-        var suggestion = findSuggestionForPhrase(phrase)
-        if suggestion == nil {
-            suggestion = NSEntityDescription.insertNewObject(forEntityName: ReaderSearchSuggestion.classNameWithoutNamespaces(),
-                                                                             into: managedObjectContext) as? ReaderSearchSuggestion
-            suggestion?.searchPhrase = phrase
+        self.coreDataStack.performAndSave { context in
```

Comment:
Nitpick: Is it necessary to access `coreDataStack` from `self` here?

```suggestion
        coreDataStack.performAndSave { context in
```